frontend/src/pages/SignUp/SignUp.tsx

```diff
@@ -38,23 +38,32 @@ type FormValues = {
 };
 
 function SignUp({ version }: SignUpProps): JSX.Element {
+	const { search } = useLocation();
+	const { notifications } = useNotifications();
 	const { t } = useTranslation(['signup']);
-	const [loading, setLoading] = useState(false);
+	const params = new URLSearchParams(search);
+	const token = params.get('token');
+
+	const [form] = Form.useForm<FormValues>();
+	const email = Form.useWatch('email', form);
```

Comment:
actually i don't to useWatch as on every keystroke it will rerender the whole form